coffeecard/CoffeeCard.Tests.Unit/Services/v2/VoucherServiceTests.cs

```diff
@@ -0,0 +1,165 @@
+using System.ComponentModel;
+using System.Linq;
+using CoffeeCard.Common.Configuration;
+using CoffeeCard.Common.Errors;
+using CoffeeCard.Library.Persistence;
+using CoffeeCard.Library.Services.v2;
+using CoffeeCard.Models.DataTransferObjects.v2.Voucher;
+using CoffeeCard.Models.Entities;
+using Microsoft.EntityFrameworkCore;
+using Xunit;
+
+namespace CoffeeCard.Tests.Unit.Services.v2
+{
+    public class VoucherServiceTests
+    {
+        [Theory (DisplayName = "CreateVouchers returns unique responses")]
+        [InlineData(0, 1)]
+        [InlineData(1, 1)]
+        [InlineData(10, 1)]
+        [InlineData(100, 1)]
+        [InlineData(50000, 1)]
+        public async void CreateVouchersReturnsUniqueResponses(int amount, int productId)
+        {
+            // Arrange
+            var builder = new DbContextOptionsBuilder<CoffeeCardContext>()
+                .UseInMemoryDatabase(nameof(CreateVouchersReturnsUniqueResponses) +
+                                     amount + productId);
+            var databaseSettings = new DatabaseSettings
+            {
+                SchemaName = "test"
+            };
+            var environmentSettings = new EnvironmentSettings()
+            {
+                EnvironmentType = EnvironmentType.Test
+            };
+
+            await using var context = new CoffeeCardContext(builder.Options, databaseSettings, environmentSettings);
+            
+            var product = new Product { Id = 1, };
+            context.Products.Add(product);
+            var voucherService = new VoucherService(context);
+
+            // Act
+            var request = new IssueVoucherRequest { Amount = amount, ProductId = productId };
+            await voucherService.CreateVouchers(request);
+            var responses = (await voucherService.CreateVouchers(request)).ToList();
+            
+            // Assert
+            Assert.True(responses.Count == amount);
```

Comment:
Avoid using Assert.True for equality comparisons, use Assert.Equal instead as this gives better output in case of failures
```charp
Assert.Equal(expected, result);
```